framework/src/executioners/Eigenvalue.C

```diff
@@ -24,7 +24,8 @@ Eigenvalue::validParams()
 {
   InputParameters params = Steady::validParams();
 
-  params.addClassDescription("Eigenvalue solves a standard/generalized eigenvaue problem");
+  params.addClassDescription(
+      "Eigenvalue solves a standard/generalized linear or nonlinear eigenvaue problem");
 
   params.addParam<bool>(
       "matrix_free",
```

Comment:
Do we still need this parameter here?

Reply:
Now I am thinking based on our discussion that we should remove `precond_matrix_includes_eigen`. Users use kernels to decide how to assemble `A` and `B`. The code should manipulate `A` and `B` with `x` and `lambda` to form the full preconditioning matrix. We possibly can detect if `B` is zero or not. If `B` is zero, the manipulation is not needed.

Reply:
matrix_free will be an invalid for griffin once the integration is done. In this PR, we can not change it.  It will still be a valid option when users  use non-Newton algorithms